src/cpu_platform.cpp

```diff
@@ -0,0 +1,41 @@
+#include "cpu_platform.h"
+#include "runtime.h"
+
+#include <algorithm>
+#include <fstream>
+#include <iterator>
+#include <string>
+
+#if defined(__APPLE__)
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#endif
+
+
+CpuPlatform::CpuPlatform(Runtime* runtime)
+    : Platform(runtime)
+{
+    #if defined(__APPLE__)
+    size_t buf_len;
+    sysctlbyname("machdep.cpu.brand_string", nullptr, &buf_len, nullptr, 0);
+    char buf[buf_len];
+    sysctlbyname("machdep.cpu.brand_string", &buf, &buf_len, nullptr, 0);
+    device_name_ = buf;
+    #elif defined(_WIN32)
+    // TODO
```

Comment:
The windows implementation is missing here.
@stlemme: can you have a look at this?

Reply:
added windows implementation